Add Cloud Storage Service - Reference Application

surfdoc opened the pull request:
Moved images files to assets/images folder
Added cloud-storage-reference-app.md with a description of the reference application we are building for the MVP Updated security.md with the change in location of the referenced image file

s1monj reviewed (changes requested):
@surfdoc I've attached an updated diagram:
 1. renamed to "Cloud Storage Service"
 2. renamed "Personal Device" from Simona's comments
 3. added consent checkboxes from Simona's comments
 
 @surfdoc "medical device" is also mentioned a few times throughout, maybe replace with "personal device" from Simona's comments

 
![css_overview](https://github.com/jupyterhealth/software-documentation/assets/16945182/9e7e998d-f5d9-4bd2-ae23-5f5b52eddafc)

surfdoc commented:
@s1monj updated to personal device when it stated medical device and also updated when it was just device to either personal device, mobile device, or Android device depending upon the context. Also removed the old diagram and added the one you attached.

ryanlovett commented on cloud-storage-reference-app.md:
```diff
@@ -0,0 +1,366 @@
+    Dr Debora->>SMART CSS: css.org<br/>Sign up / Sign in
+    opt
+        SMART CSS->>Provider Directory Service: SAML/SSO Auth
+        Provider Directory Service->>SMART CSS:
```
MyST's mermaid support needs a return message description following the `:` here, e.g. `Provider Directory Service->>SMART CSS: Auth response`.

surfdoc replied:
Same issue as above

s1monj replied:
Thanks @ryanlovett - ahhhhh I see so there is supposed to be a white space char following the ":" that must have been trimmed along the way. I just checked and that works with MyST ✅ 

@surfdoc if you replace everything between the `` ```mermaid `` with the code below it should work (and will fix the related below issue too)

```
sequenceDiagram
    actor Dr Debora
    participant SMART CSS
    participant Provider Directory Service
    participant Secondary App (SMART Client)
    Dr Debora->>SMART CSS: css.org<br/>Sign up / Sign in
    opt
        SMART CSS->>Provider Directory Service: SAML/SSO Auth
        Provider Directory Service->>SMART CSS: 
    end
    Dr Debora->>SMART CSS: Set up study, users and groups
    Dr Debora->>SMART CSS: Register Patient, create and send deep link
    Dr Debora->>Secondary App (SMART Client): Click to launch
    Secondary App (SMART Client)->>SMART CSS: SMART Auth flow
    SMART CSS->>Secondary App (SMART Client):  
    Secondary App (SMART Client)->>SMART CSS: GET /fhir/metadata
    SMART CSS->>Secondary App (SMART Client): Describe Observation valueAttachments
    Secondary App (SMART Client)->>SMART CSS: GET /observation?patient=123456
    SMART CSS->>Secondary App (SMART Client): Observation valueAttachments
    Secondary App (SMART Client)->>Dr Debora: Display shared personal device data
```

surfdoc replied:
@ryanlovett - There is a pre-commit that runs and removes trailing whitespaces. I think it is trimming these automatically.

ryanlovett replied:
@minrk What do you think about disabling the pre-commit check for trailing whitespace?

minrk replied:
Sure, no problem!

s1monj replied:
Or alternatively we can just repeat the label as per Ryan's original suggestion if that's easier? I wouldn't want you guys changing your whole pipeline just for this one diagram!

ryanlovett replied:
Would coming up with new labels or repeating the previous ones for the returns be straightforward?

If this is a needed pattern for this and future diagrams then it isn't too hard to disable the pre-commit check.

surfdoc replied:
@s1monj and @ryanlovett  - I have been summoned for jury duty this week so will have limited availability, however, let me know what you decide and I will make the appropriate changes and update the PR. Thanks.

s1monj replied:
@ryanlovett yep for sure - I'll just get rid of the second line it's not needed. Also looks like mermaid has just added double-headed arrows so will prob be available in Github in a few weeks.

@surfdoc if you can please replace with code below thanks
```
sequenceDiagram
    actor Dr Debora
    participant SMART CSS
    participant Provider Directory Service
    participant Secondary App (SMART Client)
    Dr Debora->>SMART CSS: css.org<br/>Sign up / Sign in
    opt
        SMART CSS->>Provider Directory Service: SAML/SSO Auth
        Provider Directory Service->>SMART CSS: 
    end
    Dr Debora->>SMART CSS: Set up study, users and groups
    Dr Debora->>SMART CSS: Register Patient, create and send deep link
    Dr Debora->>Secondary App (SMART Client): Click to launch
    Secondary App (SMART Client)->>SMART CSS: SMART Auth flow
    SMART CSS->>Secondary App (SMART Client):  
    Secondary App (SMART Client)->>SMART CSS: GET /fhir/metadata
    SMART CSS->>Secondary App (SMART Client): Describe Observation valueAttachments
    Secondary App (SMART Client)->>SMART CSS: GET /observation?patient=123456
    SMART CSS->>Secondary App (SMART Client): Observation valueAttachments
    Secondary App (SMART Client)->>Dr Debora: Display shared personal device data
```

surfdoc replied:
@s1monj I updated the mermaid diagrams removing lines where there was only a white space after the colon for the return arrow. I also made the requested changes if you want to re-review the PR. Thanks.

ryanlovett replied:
@surfdoc Thanks, this renders correctly in MyST now! @s1monj if it looks good I think we can merge.

ryanlovett commented on cloud-storage-reference-app.md:
```diff
@@ -0,0 +1,366 @@
+    Dr Debora->>SMART CSS: Register Patient, create and send deep link
+    Dr Debora->>Secondary App (SMART Client): Click to launch
+    Secondary App (SMART Client)->>SMART CSS: SMART Auth flow
+    SMART CSS->>Secondary App (SMART Client):
```
Same as above -- MyST wants something following `(SMART Client): `.

surfdoc replied:
@s1monj I am not sure what to do here since if I change it then I get an error. I guess we could just screen capture as an image?

ryanlovett commented on cloud-storage-reference-app.md:
```diff
@@ -0,0 +1,366 @@
+1. Provider interacts with the Jupyter Health dashboards. Initially this may be a simple web application, but will eventually be a Provider SMART app within the EHR vendor environment.
+1. A REST API with OAuth2 for programmatic access
+
+![](assets/images/chcs_iam_overview.jpg)
```
The submitted image is named `css_iam_overview.jpg` rather than `chcs_iam_overview.jpg`.

surfdoc replied:
Fixed

ryanlovett reviewed (commented):
There were a couple of minor issues preventing one of the mermaid charts from rendering, and a filename mismatch. Otherwise everything looks good!
